troubleshooting.md

```diff
@@ -916,6 +916,209 @@ After deleting a VM on macOS, the initialization of subsequent VMs fails.
+or the __--user__ option is not used.
+
+Run the container again but now with UIDs and GIDs mapped
+
+```Text
+$ subuidSize=$(( $(podman info --format "{{ range .Host.IDMappings.UIDMap }}+{{.Size }}{{end }}" ) - 1 ))
+$ subgidSize=$(( $(podman info --format "{{ range .Host.IDMappings.GIDMap }}+{{.Size }}{{end }}" ) - 1 ))
+$ mkdir dir1
+$ chmod 777 dir1
+$ podman run --rm
+  -v ./dir1:/dir1:Z \
+  --user $uid:$gid \
+  --uidmap $uid:0:1 \
+  --uidmap 0:1:$uid \
+  --uidmap $(($uid+1)):$(($uid+1)):$(($subuidSize-$uid)) \
+  --gidmap $gid:0:1 \
+  --gidmap 0:1:$gid \
+  --gidmap $(($gid+1)):$(($gid+1)):$(($subgidSize-$gid)) \
+     docker.io/library/ubuntu bash -c "touch /dir1/a; chmod 600 /dir1/a"
+$ id -u
+tester
+$ id -g
+tester
+$ ls -l dir1/a
+-rw-------. 1 tester tester 0 Jan 19 20:31 dir1/a
+$ 
+```
+
+In this example the __--user__ option specified a rootless user in the container.
+As the rootless user could also have been specified in the container image, e.g.
+
+```Text
+$
+podman image inspect --format "user: {{.User}}" IMAGE
+user: hpc
+$
+```
+, the same problem could also occur even without specifying __--user=__.
+
+Another variant of the same problem could occur when using
+__--user=root:root_ (the default), but where the root user creates non-root owned files
+in some way (e.g by creating them themselves, or switching the effective UID to
+a rootless user and then creates files).
+
+Using `--userns=keep-id --user $(id -u):$(id -g)` could be an alternative solution. The option
+[__--userns=keep-id__](https://docs.podman.io/en/latest/markdown/podman-run.1.html#userns-mode) also creates
+a similar UID mapping and GID mapping but for the special case when the container UID is equal
+to the user's regular UID on the host.
+
+### 33) Passed-in devices or files can't be accessed in rootless container
+
+As a non-root user you have access rights to devices, files and directories that you
+want to pass into a rootless container with `--device=...`, `--volume=...` or `--mount=..`.
+
+Podman by default maps a non-root user inside a container to one of the user's
+subordinate UIDs and subordinate GIDs on the host. When the container user tries to access a 
+file, a "Permission denied" error could occur because the container user does not have the
+permissions of the regular user of the host. 
+
+#### Symptom
+
+* Any access inside the container is rejected with "Permission denied"
+for files, directories or devices passed in to the container
+with `--device=..`,`--volume=..` or `--mount=..`, e.g.
+
+```Text
+$ mkdir dir1
+$ chmod 700 dir1
+$ podman run --rm -v ./dir1:/dir1:Z \
+             --user 2003:2003 \
+             docker.io/library/ubuntu ls /dir1
+ls: cannot open directory '/dir1': Permission denied
+```
+
+#### Solution
+
+We follow essentialy the same solution as in the previous
+troubleshooting tip
+ "32) Container creates a file that is not owned by the regular UID"
```

Comment:
We have renumbered these in the past, I'd drop the number and just use the title.
```suggestion
 "Container creates a file that is not owned by the regular UID"
```